tests/integration_test.rs

```diff
@@ -0,0 +1,217 @@
+    None,
+    false
+)]
+#[case::wasi(
+    PolicyExecutionMode::Wasi,
+    "ghcr.io/kubewarden/tests/go-wasi-template:v0.1.0",
+    json!({
+        "requiredAnnotations": {
+            "fluxcd.io/cat": "felix"
+        }
+    }),
+    "pod_creation_flux_cat.json",
+    false,
+    true,
+    None,
+    None,
+    false
+)]
+#[case::wasi_mutating(
+    PolicyExecutionMode::Wasi,
+    "ghcr.io/kubewarden/tests/go-wasi-template:v0.1.0",
+    json!({
+        "requiredAnnotations": {
+            "fluxcd.io/cat": "felix"
+        }
+    }),
+    "service_clusterip.json",
+    false,
+    true,
+    None,
+    None,
+    true
+)]
+#[case::wapc_raw(
+    PolicyExecutionMode::KubewardenWapc,
+    "ghcr.io/kubewarden/tests/raw-validation-policy:v0.1.0",
+    json!({
+        "validUsers": ["tonio", "wanda"],
+        "validActions": ["eats", "likes"],
+        "validResources": ["banana", "hay"],
+    }),
+    "raw_validation.json",
+    true,
+    true,
+    None,
+    None,
+    false
+)]
+#[case::wapc_raw_mutating(
+    PolicyExecutionMode::KubewardenWapc,
+    "ghcr.io/kubewarden/tests/raw-mutation-policy:v0.1.0",
+    json!({
+        "forbiddenResources": ["banana", "carrot"],
+        "defaultResource": "hay",
+    }),
+    "raw_mutation.json",
+    true,
+    true,
+    None,
+    None,
+    true
+)]
+#[case::opa_raw(
+    PolicyExecutionMode::Opa,
+    "ghcr.io/kubewarden/tests/raw-validation-opa-policy:v0.1.0",
+    json!({}),
+    "raw_validation.json",
+    true,
+    true,
+    None,
+    None,
+    false
+)]
+#[case::wasi_raw(
+    PolicyExecutionMode::Wasi,
+    "ghcr.io/kubewarden/tests/raw-validation-wasi-policy:v0.1.0",
+    json!({
+        "validUsers": ["tonio", "wanda"],
+        "validActions": ["eats", "likes"],
+        "validResources": ["banana", "hay"],
+    }),
+    "raw_validation.json",
+    true,
+    true,
+    None,
+    None,
+    false
+)]
+#[case::wasi_raw_mutating(
+    PolicyExecutionMode::Wasi,
+    "ghcr.io/kubewarden/tests/raw-mutation-wasi-policy:v0.1.0",
+    json!({
+        "forbiddenResources": ["banana", "carrot"],
+        "defaultResource": "hay",
+    }),
+    "raw_mutation.json",
+    true,
+    true,
+    None,
+    None,
+    true
+)]
+#[tokio::test]
+async fn test_policy_evaluator(
+    #[case] execution_mode: PolicyExecutionMode,
+    #[case] policy_uri: &str,
+    #[case] settings: serde_json::Value,
+    #[case] request_file_path: &str,
+    #[case] raw: bool,
+    #[case] allowed: bool,
+    #[case] message: Option<String>,
+    #[case] code: Option<u16>,
+    #[case] mutating: bool,
+) {
+    let tempdir = tempfile::TempDir::new().expect("cannot create tempdir");
+    let policy = policy_evaluator::policy_fetcher::fetch_policy(
+        policy_uri,
+        PullDestination::LocalFile(tempdir.into_path()),
+        None,
+    )
+    .await
+    .expect("cannot fetch policy");
+
+    let data = std::fs::read(policy.local_path).expect("cannot read policy file");
```

Comment:
You can skip that and let the builder read the contents via the `.policy_file(path)` method